### ISSUE TITLE
Rewrite references to channels

### ISSUE THREAD

**Issue by sporksmith** [matrix-org/matrix-appservice-irc#1304]
**Is your feature request related to a problem? Please describe.**

As a Matrix user in a bridged room, if I mention an IRC channel, IRC users see the name of the corresponding Matrix room. This seems to happen even if I don't tab-complete to make a "pilled" mention when composing.

e.g. "Please take this to #tor-offtopic" is seen by IRC users as "Please take this to #_oftc_#tor-offtopic:matrix.org"

**Describe the solution you'd like**

It'd be nice if mentions of channels got rewritten to and from their portal Matrix room counterparts automatically. I imagine this could be done with relatively simple pattern matching.

It'd be even nicer if this also worked with mentions of Matrix rooms that are plumbed to an IRC room. e.g. #tor:matrix.org with #tor. I imagine that'd require using a dynamic map, which could be a bit more complex. OTOH perhaps it'd also cut down on false positives/rewrites vs simple pattern-matching.

**Comment by Half-Shot** [comment]
I've renamed this to make the feature request clear, it should be pretty possible to do this.

**Comment by sporksmith** [comment]
FWIW my original title was going to be something similar, but then I tried to follow the guideline from your [CONTRIBUTING.md](https://github.com/matrix-org/matrix-appservice-irc/blob/develop/CONTRIBUTING.md) "Write a short title which neatly summaries the problem. Do not write the solution in the issue title. ". OTOH maybe you as the developer replacing the "problem" issue title with what *you* consider the solution to be is the intended workflow :)

**Comment by Half-Shot** [comment]
Yeah, you did the right thing :+1: .  I'm just renaming for our own benefit so we can instantly see what work needs to be done. It's not uncommon for an issue to be opened with a problem description that is then renamed once we've come up with something appropriate to fix it with.

**Comment by tgr** [comment]
A related issue is when a Matrix room linked to an IRC room is mentioned (and the bot knows about the link). E.g. `#a:matrix.org` gets bridged to `#a` on some IRC network, `#b:matrix.org` gets bridged to `#b` on the same IRC network by the same bot, some Matrix uses mentions `#b:matrix.org` in the `#a:matrix.org` room. The bot should have every information in this situation to translate it as `#b` for IRC users.
